R/license_note.R

```diff
@@ -0,0 +1,77 @@
+#' Generate LICENSE.note file.
+#'
+#' LICENSE.note generated by this function lists dependent Rust packages infomation.
+#' To use this function, the [cargo-lincense](https://crates.io/crates/cargo-license) command must be installed.
+#' @param force Logical indicating whether to re-generating LICENSE.note if LICENSE.note is already exists.
+#' @inheritParams register_extendr
+#' @return No return value, called for side effects.
+#' @export
+#' @importFrom dplyr %>%
+write_license_note <- function(path = ".", force = TRUE) {
+  if (!requireNamespace("RcppTOML", quietly = TRUE)) {
+    cli::cli_abort(
+      "The {.pkg RcppTOML} package is required to run the {.fun write_license_note} function.",
+      class = "rextendr_error"
+    )
+  }
+
+  manifest_file <- rprojroot::find_package_root_file("src", "rust", "Cargo.toml", path = path)
+  out_file <- rprojroot::find_package_root_file("LICENSE.note", path = path)
+
+  if (!isTRUE(force) && fs::file_exists(out_file)) {
+    cli::cli_abort(
+      c(
+        "LICENSE.note already exists.",
+        "If you want to re-generate LICENSE.note, set `force = TRUE` to {.fun write_license_note}."
+      ),
+      class = "rextendr_warning"
+    )
+  }
+
+  note_header <- paste0(
+    "The binary compiled from the source code of this package contains the following Rust packages.\n",
+    "\n",
+    "\n",
+    "-------------------------------------------------------------"
```

Comment:
Could you use `cli::console_width()` instead of a predefined number of `----`? That way users don't run into undesireable wrapping. 

Alternatively `cli::cli_rule()` handles width for you.